src/status_im2/contexts/chat/lightbox/top_view.cljs

```diff
@@ -39,14 +42,33 @@
         (anim/animate top-view-width screen-width)
         (anim/animate top-view-bg colors/neutral-100-opa-0)))))
 
+(defn drawer
+  [uri]
+  [quo/action-drawer
+   [[{:icon                :i/save
+      :accessibility-label :save-image
+      :label               (i18n/label :t/save-image-library)
+      :on-press            (fn []
+                             (rf/dispatch [:hide-bottom-sheet])
+                             (images/save-image-to-gallery
+                              uri
+                              #(rf/dispatch [:toasts/upsert
+                                             {:id              :random-id
```

Comment:
beyond the scope of this pr:
why do we even need id's if we're going to use them like this? would it not be better to just have the toast create a random id when one is not provided?